types/tx.go

```diff
@@ -1,3 +1,24 @@
 package types
 
+import (
+	"github.com/tendermint/go-merkle"
+)
+
 type Tx []byte
+
+type Txs []Tx
+
+func (txs Txs) Hash() []byte {
```

Comment:
Also, [][]byte allocation doesn't seem like it would matter much for performance, considering this only happens once per block.  So SimpleHashFromBinaries seems like it should be fine?

```golang
package main

import (
	"fmt"
	"testing"
)

func BenchmarkArraysCreation(b *testing.B) {
	total := 0
	for i := 0; i < b.N; i++ {
		foo := make([][]byte, 10000)
		total += len(foo)
	}
	fmt.Println(total)
}
```

prints

```
BenchmarkArraysCreation-4
   30000	     48357 ns/op
ok  	command-line-arguments	1.999s
```